hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -480,6 +480,25 @@ def __getitem__(self, item):
         else:
             return self._method("indexArray", self.dtype.element_type, item)
 
+    def _slice(self, ret_type, start=None, stop=None, step=None):
+        if step is None:
+            step = 1
+        if step >= 0:
+            if start is None:
+                start = 0
```

Comment:
This is not allowed. `step` can be a `Int32Expression`, so you can't just compare it like this. If `step` is an `Int32Expression`, then `step >= 0` will return a `BooleanExpression`, which will cause a crash since `if` can't handle that.

Reply:
`start = hl.if_else(step >= 0, ............., ...........)`